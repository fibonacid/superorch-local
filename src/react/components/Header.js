import React from 'react';
import * as styled from './styles/Header.styles';
import PropTypes from 'prop-types';

function Header(props) {
  return (
    <styled.Container data-test={'HeaderComponent'}>
      <styled.Title data-test={'title'}>
        {props.appName}
      </styled.Title>
    </styled.Container>
  )
}

Header.propTypes = {
  appName: PropTypes.string
};

export default Header;
