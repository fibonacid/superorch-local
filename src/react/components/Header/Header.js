import React from 'react';
import styled from 'styled-components/macro'
import PropTypes from 'prop-types';

const StyledContainer = styled.div`
  font-size: 18px;
  font-weight: bold;
  padding: 5px;
  flex: 0 0 auto;
  display: flex;
  justify-content: center;
`;

const StyledTitle = styled.h1`
  /*text-align: center;*/ 
`;

function Header(props) {
  return (
    <StyledContainer data-test={'HeaderComponent'}>
      <StyledTitle data-test={'title'}>
        {props.appName || "ColliderChat"}
      </StyledTitle>
    </StyledContainer>
  )
}

Header.propTypes = {
  appName: PropTypes.string
};

export default Header;
