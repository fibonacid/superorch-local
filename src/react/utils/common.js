import _ from "lodash";

/**
 * This function is used to transform
 * all the prop keys to lower case.
 * By doing so you can avoid errors like this:
 *
 * React does not recognize the `contentState` prop on a DOM element.
 * If you intentionally want it to appear in the DOM as a custom attribute,
 * spell it as lowercase `contentstate` instead.
 * If you accidentally passed it from a parent component, remove it from the DOM element.
 *
 * @param props
 * @returns {{}}
 */
export function lowerCasedProps(obj) {
  return _.mapKeys(obj, (value, key) => {
    return key.toLowerCase()
  });
}
