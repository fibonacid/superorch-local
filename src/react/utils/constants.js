export const INITIAL_USER_ID = -1;
