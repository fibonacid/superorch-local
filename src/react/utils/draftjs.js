import { OrderedMap, List, Map } from "immutable";

/**
 * Function returns collection of currently selected blocks.
 */
export function getSelectedBlocksMap(editorState: EditorState): OrderedMap {
  const selectionState = editorState.getSelection();
  const contentState = editorState.getCurrentContent();
  const startKey = selectionState.getStartKey();
  const endKey = selectionState.getEndKey();
  const blockMap = contentState.getBlockMap();
  return blockMap
    .toSeq()
    .skipUntil((_, k) => k === startKey)
    .takeUntil((_, k) => k === endKey)
    .concat([[endKey, blockMap.get(endKey)]]);
}

/**
 * Function returns collection of currently selected blocks.
 */
export function getSelectedBlocksList(editorState: EditorState): List {
  return getSelectedBlocksMap(editorState).toList();
}

export function getSelectedTextBlocks(editorState: EditorState): string {
  const selectedBlocks = getSelectedBlocksList(editorState);
  let text;
  selectedBlocks.forEach((block, i) => {
    text = `${block.getText()}${
      i < selectedBlocks.size
        ? '\n'
        : ''
    }`;
  });
  return text;
}

export function findWithRegex(regex, contentBlock, callback) {
  const text = contentBlock.getText();
  let matchArr, start;
  while ((matchArr = regex.exec(text)) !== null) {
    start = matchArr.index;
    callback(start, start + matchArr[0].length);
  }
}
