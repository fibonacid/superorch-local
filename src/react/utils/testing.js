import { checkPropTypes } from "prop-types";

export const testFunction = () => {};

export const findByTestAttr = (component, attr) => {
  const wrapper = component.find(`[data-test='${attr}']`);
  return wrapper;
};

export const checkProps = (component, expectedProps) => {
  const propsErr = checkPropTypes(
    component.propTypes,
    expectedProps,
    "props",
    component.name
  );
  return propsErr;
};
