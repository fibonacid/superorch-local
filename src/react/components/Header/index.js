import { connect } from "react-redux";
import Header from "./Header";

const mapStateToProps = state => ({
  appName: state.base.name
});

export default connect(mapStateToProps, null)(Header);
