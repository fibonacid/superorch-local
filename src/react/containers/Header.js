import {connect} from "react-redux";
import Header from "../components/Header";

const mapStateToProps = state => ({
  appName: state.app.name
});

export default connect(mapStateToProps, null)(Header);
