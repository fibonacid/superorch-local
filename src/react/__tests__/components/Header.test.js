import React from "react";
import {shallow} from 'enzyme'
import Header from "../../components/Header/Header";
import {findByTestAttr} from "../../utils/testing";
import toJson from "enzyme-to-json";

const setUp = (props={}) => {
  return shallow(<Header {...props} />);
};

describe('Header Component', () => {

  describe('When has props', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = setUp({ appName: "test"} )
    });
    afterEach(() => {
      wrapper.unmount();
    });
    it('Should renderer without errors', () => {
      const component = findByTestAttr(wrapper, 'HeaderComponent');
      expect(component.length).toBe(1);
    });
    it('Should have a title', () => {
      const title = findByTestAttr(wrapper, 'title');
      expect(title.text()).toBeTruthy();
    })
  });

  describe('When has NO props', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = setUp()
    });
    afterEach(() => {
      wrapper.unmount();
    });
    it('Should not render', () => {}).todo();
  });
});
