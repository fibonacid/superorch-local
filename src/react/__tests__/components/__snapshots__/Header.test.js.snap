// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component When has props Should renderer without errors 1`] = `
<Headerstyles__Container
  data-test="HeaderComponent"
>
  <Headerstyles__Title
    data-test="title"
  >
    test
  </Headerstyles__Title>
</Headerstyles__Container>
`;
